### PR TITLE
HA dqlite

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

**Comment by joeky888** [comment]
Could you provide some benchmarks of using dqlite and etcd? Thanks!

File: snapcraft.yaml [hunk 1]
```diff
@@ -280,20 +280,19 @@ parts:
 
       REMOVE_KUBE_SNAP_BINS=false
       echo "Reached here"
+      # if "${KUBE_SNAP_BINS}" exist we have to use the binaries from there
+      # if "${KUBE_SNAP_BINS}" does not exist but it is set we will put the k8s binaries there
+      # if "${KUBE_SNAP_BINS}" does not exist and it is not set we do not need to keep the created binaries
       if [ ! -e "${KUBE_SNAP_BINS}" ]; then
         if [ -z "${KUBE_SNAP_BINS}" ]; then
+          # "${KUBE_SNAP_BINS}" not set, we will remoove the binaries after we build them.
```
[the remark this answers is not in the document]
Comment (ktsakalozos) [reply snapcraft.yaml]: Thanks!

File: build-scripts/build-k8s-binaries.sh [hunk 2]
```diff
@@ -1,20 +1,34 @@
 #!/bin/bash
-set -eu
+set -eux
 
-echo "Building k8s bianries from $KUBERNETES_REPOSITORY commit $KUBERNETES_COMMIT"
+echo "Building k8s bianries from $KUBERNETES_REPOSITORY tag $KUBERNETES_TAG"
```
[the remark this answers is not in the document]
Comment (ktsakalozos) [reply build-scripts/build-k8s-binaries.sh]: Thanks

**Comment by balchua** [comment]
What do you think about this ? https://github.com/rancher/kine an etcdshim, to avoid patching the apiserver.

**Comment by ktsakalozos** [comment]
> What do you think about this ? https://github.com/rancher/kine

We practically use this code patched to be embedded into the apiserver.

**Comment by ktsakalozos** [comment]
This work has moved to https://github.com/ubuntu/microk8s/pull/1208

**Pull request closed without merging**